apps/roam/src/components/CreateRelationDialog.tsx

```diff
@@ -71,33 +69,45 @@ const CreateRelationDialog = ({
 
   const relKeys = Object.keys(relDataByTag);
   const [selectedRelationName, setSelectedRelationName] = useState(relKeys[0]);
-  const [selectedTargetTitle, setSelectedTargetTitle] = useState<string>("");
-  const [selectedTargetUid, setSelectedTargetUid] = useState<
-    string | undefined
-  >(undefined);
-  const allPages = useMemo(() => getAllPageNames().sort(), []);
-  const getFilteredPageNames = (selectedRelationName: string): string[] => {
+  const [selectedTarget, setSelectedTarget] = useState<Result>({
+    text: "",
+    uid: "",
+  });
+  const pageUidByTitle = useMemo(
+    () =>
+      Object.fromEntries(
+        window.roamAlphaAPI.data.fast.q(
+          "[:find ?t ?u :where [?p :node/title ?t] [?p :block/uid ?u]]",
+        ) as [string, string][],
+      ),
+    [],
+  );
+  const allPages = useMemo(
```

Comment:
nit: we don't need two useMemo's and multiple passes, eg

```typescript
const allPages = useMemo(() => {
  const rows = window.roamAlphaAPI.data.fast.q(
    "[:find ?t ?u :where [?p :node/title ?t] [?p :block/uid ?u]]",
  ) as [string, string][];

  return rows.map(([text, uid]): Result => ({ uid, text }));
}, []);
```

Reply:
Actually, I use both variables separately, so I think it's worth having two useMemos.